ckanext/datastore/plugin.py

```diff
@@ -282,21 +282,9 @@ def before_show(self, resource_dict):
                 controller='ckanext.datastore.controller:DatastoreController',
                 action='dump', resource_id=resource_dict['id'])
 
-        connection = None
+        if 'datastore_active' not in resource_dict:
+            resource_dict[u'datastore_active'] = False
```

Comment:
Correction: It's not this part, it's simply the fact that we have an IResourceController plugin enabled. https://github.com/ckan/ckan/blob/master/ckan/logic/action/get.py#L998 just looping over the resources to pass them to IResourceController.before_show() causes our LazyJSONObject to not be lazy any more.

Since so many sites enable the datastore maybe it's not worth having the LazyJSONObject at all. Or we could only apply the before_show calls on unvalidated data_dicts (and cache the value along with the rest of the metadata).